src/python/pants/init/extension_loader.py

```diff
@@ -136,22 +136,38 @@ def load_backend(build_configuration: BuildConfiguration.Builder, backend_packag
     :raises: :class:``pants.base.exceptions.BuildConfigurationError`` if there is a problem loading
       the build configuration.
     """
-    backend_module = backend_package + ".register"
+
+    if templating_args:
+        kwargs = {'backend_package': backend_package}
+        kwargs.update(templating_args)
+        backend_module = kwargs.pop('template') + ".register"
+    else:
+        kwargs = {}
+        backend_module = backend_package + ".register"
```

Comment:
I always try to consolidate the "end result" so in this case we only have a single expression for the `backend_module`, something like (also incorporated my other suggestions here):
```suggestion
    if templating_args:
        kwargs = {'backend_package_alias': backend_package}
        kwargs.update(templating_args)
        backend_package = kwargs.pop('backend_package')
    else:
        kwargs = {}
    backend_module = backend_package + ".register"
```

Reply:
Currently it looks like this:
```python
    if templating_config:
        kwargs = {'backend_package_alias': backend_package}
        kwargs.update(templating_config.kwargs)
        backend_module = templating_config.package + ".register"
    else:
        kwargs = {}
        backend_module = backend_package + ".register"
```
The quirk that this is attempting to accommodate which would be harder if we do:
```python
    if templating_config:
        kwargs = {'backend_package_alias': backend_package}
        kwargs.update(templating_config.kwargs)
        backend_package = templating_config.package
    else:
        kwargs = {}
    
    backend_module = backend_package + ".register"
```
is that during registration we use `backend_package` to label a kind of provider for each target or rule:
```python
if target_types:
    build_configuration.register_target_types(backend_package, target_types)
```
I imagine that here it would be useful to be passing in the "alias" (the member of `GLOBAL.backend_packages`) rather than the package used to template out several backends. That way each Target and rule could be tracked back not just to the template but to its specific instantiation.

Reply:
Would it be better to refer to the templating package instead?

Reply:
I've changed the code in the area significantly, but it does now have a single expression to produce the output of the conditional, so marking as resolved.